test/jsonapi/plugs/query_parser_test.exs

```diff
@@ -108,6 +108,16 @@ defmodule JSONAPI.QueryParserTest do
     end
   end
 
+  test "parse_include/2 errors with limited allowed includes" do
```

Comment:
I was playing with this. I'm not sure it's ideal, but I also don't think it's a blocker. We'll need to include-list all combinations of nested includes as this stands. Below, I added a suggestion to explicitly test the nested includes case to illustrate this.

```suggestion
test "parse_include/2 errors with limited allowed includes" do
    config = struct(Config, view: MyView, opts: [include: ~w(author comments comments.user)])

    assert_raise InvalidQuery, "invalid include, best_friends for type mytype", fn ->
      parse_include(config, "best_friends,author")
    end

    assert parse_include(config, "author,comments").include == [:author, :comments]

    assert parse_include(config, "author,comments.user").include == [:author, {:comments, :user}]
  end
```